src/alchemlyb/tests/test_preprocessing.py

```diff
@@ -10,117 +10,284 @@
+
+    @pytest.mark.parametrize(('data', 'size', 'column'),
+                             [(gmx_benzene_dHdl(), 20005, 0),
+                              (gmx_benzene_u_nk(), 20005, 0)])
+    def test_subsampling_series(self, data, size, column):
+        assert len(self.subsampler(data, column=data[data.columns[column]])) <= size
+
+    def test_subsampling_dHdl(self, dHdl):
+        data, nsims = dHdl
+
+        if nsims == "single":
+            dHdl_s = self.subsampler(data, how='sum')
+            assert len(dHdl_s) < len(data)
+        elif nsims == "repeat":
+            with pytest.raises(KeyError):
+                dHdl_s = self.subsampler(data, how='sum')
+
+    def test_subsampling_u_nk(self, u_nk):
+        data, nsims = u_nk 
+
+        if nsims == "single":
+            u_nk_s = self.subsampler(data, how='right')
+            assert len(u_nk_s) < len(data)
+        elif nsims == "repeat":
+            with pytest.raises(KeyError):
+                u_nk_s = self.subsampler(data, how='right')
+
+    def test_subsampling_u_nk_left(self, u_nk):
+        data, nsims = u_nk 
+
+        if nsims == "single":
+            u_nk_s = self.subsampler(data, how='left')
+            assert len(u_nk_s) < len(data)
+        elif nsims == "repeat":
+            with pytest.raises(KeyError):
+                u_nk_s = self.subsampler(data, how='left')
+
+    def test_subsampling_u_nk_random(self, u_nk):
+        data, nsims = u_nk 
+
+        if nsims == "single":
+            u_nk_s = self.subsampler(data, how='random', random_state=42)
+            assert len(u_nk_s) < len(data)
+        elif nsims == "repeat":
+            with pytest.raises(KeyError):
+                u_nk_s = self.subsampler(data, how='random', random_state=42)
+
+
+class TestStatisticalInefficiency(CorrelatedPreprocessors):
+
+    def subsampler(self, *args, **kwargs):
+        return statistical_inefficiency(*args, **kwargs)
 
-    @pytest.mark.parametrize('data', [gmx_benzene_dHdl(),
-                                      gmx_benzene_u_nk()])
-    def test_no_series(self, data):
-        """Check that we get the same result as simple slicing with no Series.
+    @pytest.mark.parametrize(('conservative', 'data', 'size', 'how'),
+                             [
+                                 (True, gmx_benzene_dHdl(), 2001, 'sum'),  # 0.00:  g = 1.0559445620585415
+                                 (True, gmx_benzene_u_nk(), 2001, 'right'),  # 'fep': g = 1.0560203916559594
+                                 (False, gmx_benzene_dHdl(), 3789, 'sum'),
+                                 (False, gmx_benzene_u_nk(), 3788, 'right'),
+                             ])
+    def test_conservative(self, data, size, conservative, how):
+        sliced = self.subsampler(data, how=how, conservative=conservative)
+        # results can vary slightly with different machines
+        # so possibly do
+        # delta = 10
+        # assert size - delta < len(sliced) < size + delta
```

Comment:
You could also turn it into a floating point comparison, e.g., `size/len(sliced)` as the fraction and then just use `assert_almost_equal(...., decimals=2)` or whatever. Either way, if you already know that there can be variation then put it in the test and comment it.